Library.java

```diff
@@ -0,0 +1,249 @@
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("serial")
+public class Library implements Serializable {
+	
+	private static final String LIBRARY_FILE = "library.obj";
+	private static final int LOAN_LIMIT = 2;
+	private static final int LOAN_PERIOD = 2;
+	private static final double FINE_PER_DAY = 1.0;
+	private static final double MAX_FINES_OWED = 1.0;
+	private static final double DAMAGE_FEE = 2.0;
+	
+	private static Library library;
+	private int bookID;
+	private int memberID;
+	private int loanID;
+	private Date loanDate;
+	
+	private Map<Integer, Book> catalogue;
+	private Map<Integer, Member> members;
+	private Map<Integer, loan> loans;
+	private Map<Integer, loan> currentLoans;
+	private Map<Integer, Book> damagedBooks;
+	
+
+	private Library() {
+		catalogue = new HashMap<>();
+		members = new HashMap<>();
+		loans = new HashMap<>();
+		currentLoans = new HashMap<>();
+		damagedBooks = new HashMap<>();
+		bookID = 1;
+		memberID = 1;		
+		loanID = 1;		
+	}
+
+	
+	public static synchronized Library getInstance() {		
+		if (library == null) {
+			Path objectFile = Paths.get(LIBRARY_FILE);			
+			if (Files.exists(objectFile)) {	
+				try (ObjectInputStream libraryState = new ObjectInputStream(new FileInputStream(LIBRARY_FILE));) {
+			    
+					library = (Library) libraryState.readObject();
+					Calendar.getInstance().setDate(library.loanDate);
+					libraryState.close();
+				}
+				catch (Exception e) {
+					throw new RuntimeException(e);
+				}
+			}
+			else library = new Library();
+		}
+		return library;
+	}
+
+	
+	public static synchronized void saveState() {
+		if (library != null) {
+			library.loanDate = Calendar.getInstance().getDate();
+			try (ObjectOutputStream libraryState = new ObjectOutputStream(new FileOutputStream(LIBRARY_FILE));) {
+				libraryState.writeObject(library);
+				libraryState.flush();
+				libraryState.close();	
+			}
+			catch (Exception e) {

```

Comment:
Variable e can be expanded to be something more meaningful e.g.
```suggestion
			catch (Exception exception) {
```